### PR TITLE
Icons update

### PR DESCRIPTION
**Pull request by Duncid**
- Adding the Darkmode font to be used in the UI
- Aligning tailwind conf of front and sparkle (colors and font)
- Adding a new "size" prop to Icon component
- Making use of the "size" prop in Button, IconButton, IconToggleButton
- Adding a "tooltipPosition" prop to Button, IconButton, IconToggleButton

Note:
I don't know why the package.json has changed.

### CODE REVIEW

File: sparkle/src/components/Icon.tsx [hunk 1]
```diff
@@ -1,10 +1,25 @@
-import React, { ComponentType } from "react";
+import React, { ComponentType, CSSProperties } from "react";
+import { classNames } from "@sparkle/lib/utils";
 
-interface IconProps {
+export interface IconProps {
   IconComponent?: ComponentType<{ className?: string }>;
```
Comment (spolu) [inline sparkle/src/components/Icon.tsx:5]: Isn't that a bit verbose as Name? Will that mean we have to use `IconComponent={...}` anywhere we use an icon outside of the design system, right? That does not happen that often that being said...

File: sparkle/src/styles/global_with_tw_base.css [hunk 2]
```diff
@@ -1,3 +1,4 @@
 @tailwind base;
 @tailwind components;
-@tailwind utilities;
\ No newline at end of file
+@tailwind utilities;
+@import url("https://use.typekit.net/jnb2umy.css");
```
Comment (spolu) [inline sparkle/src/styles/global_with_tw_base.css:4]: What is this?

**Review by spolu — COMMENTED** [review]
Left 2 questions, happy to help fix the warnings and build failures.

File: sparkle/src/styles/global_with_tw_base.css (hunk 2 above)
Comment (Duncid) [reply sparkle/src/styles/global_with_tw_base.css:4]: The Darkmode font used in the UI

**Comment by Duncid** [comment]
> Left 2 questions, happy to help fix the warnings and build failures.

Happy to learn what I need to do to avoid them.
Any thing I need to read to understand?

File: sparkle/src/styles/global_with_tw_base.css (hunk 2 above)
Comment (spolu) [reply sparkle/src/styles/global_with_tw_base.css:4]: We probably need to add it to `front` as well no?

**Comment by spolu** [comment]
Some of them are reported in the code section of this pull request. Mostly imports ordering and unused imports. Then you can click on Details next to "Build & Lint (sparkle)" failure message right below, and nothing else pop ups than these other warnings so that must be all of them :+1:

File: sparkle/src/styles/global_with_tw_base.css (hunk 2 above)
Comment (Duncid) [reply sparkle/src/styles/global_with_tw_base.css:4]: Yes we do.

**Comment by Duncid** [comment]
@spolu I get a new error that I do not understand.
I'd love your help.

File: front/package.json [hunk 3]
```diff
@@ -14,6 +14,7 @@
   },
   "dependencies": {
     "@dust-tt/sparkle": "0.1.29",
+    "@esbuild/darwin-arm64": "^0.19.2",
```
Comment (spolu) [inline front/package.json:17]: @Duncid the error is likely caused by that which should not be here :+1:

**Review by spolu — CHANGES_REQUESTED** [review]

File: sparkle/src/components/Icon.tsx (hunk 1 above)
Comment (spolu) [reply sparkle/src/components/Icon.tsx:6]: Roger that :+1:

**Pull request merged**